extensions/github/src/helpers/repository.ts

```diff
@@ -34,15 +34,59 @@ export const WEB_IDES = [
   },
 ];
 
-import { LocalStorage } from "@raycast/api";
+import { execSync } from "node:child_process";
+import { existsSync } from "node:fs";
+import { homedir } from "node:os";
+
+import { LocalStorage, Toast, getPreferenceValues, showToast } from "@raycast/api";
 import { useCachedState } from "@raycast/utils";
 import { useEffect } from "react";
 
 import { ExtendedRepositoryFieldsFragment } from "../generated/graphql";
 
+import { getErrorMessage } from "./errors";
+
 const VISITED_REPOSITORIES_KEY = "VISITED_REPOSITORIES";
 const VISITED_REPOSITORIES_LENGTH = 25;
 
+export async function cloneAndOpen(repository: ExtendedRepositoryFieldsFragment) {
+  const { application, baseClonePath } = getPreferenceValues<Preferences.SearchRepositories>();
+  const applicationPath = application.path.replaceAll(" ", "\\ ");
+  const clonePath = `${baseClonePath}/${repository.nameWithOwner}`;
+  const openCommand = `open -a ${applicationPath} ${clonePath}`;
+
+  const toast = await showToast({
+    title: `Opening ${repository.nameWithOwner}`,
+    message: `at ${clonePath}`,
+    style: Toast.Style.Animated,
+  });
+
+  if (!existsSync(clonePath.replace("~", homedir()))) {
+    const cloneUrl = `https://github.com/${repository.nameWithOwner}`;
+    const cloneCommand = `git clone ${cloneUrl} ${clonePath}`;
+
+    try {
+      execSync(cloneCommand);
```

Comment:
Missed that one, but you probably want to show a success toast when it's done. Otherwise, you'll get the animated toast infinitely.

Reply:
Updated, thanks.